js/packages/wallet-standard-mobile/src/wallet.ts

```diff
@@ -509,3 +511,459 @@ export class SolanaMobileWalletAdapterWallet implements Wallet {
+                // TODO: Evaluate whether there's any threat to not `awaiting` this expression
+                Promise.all([
+                    this.#authorizationResultCache.set(authorizationResult),
+                    this.#handleAuthorizationResult(authorizationResult),
+                ]);
+
+                return authorizationResult;
+            });
+        } catch (e) {
+            throw new WalletConnectionError((e instanceof Error && e.message) || 'Unknown error', e);
+        }
+    }
+
+    #handleAuthorizationResult = async (authorizationResult: AuthorizationResult) => {
+        const didPublicKeysChange =
+            // Case 1: We started from having no authorization.
+            this.#authorizationResult == null ||
+            // Case 2: The number of authorized accounts changed.
+            this.#authorizationResult?.accounts.length !== authorizationResult.accounts.length ||
+            // Case 3: The new list of addresses isn't exactly the same as the old list, in the same order.
+            this.#authorizationResult.accounts.some(
+                (account, ii) => account.address !== authorizationResult.accounts[ii].address,
+            );
+        this.#authorizationResult = authorizationResult;
+        if (didPublicKeysChange) {
+            const nextSelectedAddress = await this.#addressSelector.select(
+                authorizationResult.accounts.map(({ address }) => address),
+            );
+            if (nextSelectedAddress !== this.#selectedAddress) {
+                this.#selectedAddress = nextSelectedAddress;
+                this.#emit('change',{ accounts: this.accounts });
+            }
+        }
+    }
+
+    #performReauthorization = async (wallet: Web3MobileWallet, authToken: AuthToken) => {
+        try {
+            const mwaAuthorizationResult = await wallet.authorize({
+                auth_token: authToken,
+                identity: this.#appIdentity,
+            });
+            
+            const accounts = this.#accountsToWalletStandardAccounts(mwaAuthorizationResult.accounts)
+            const authorizationResult = { ...mwaAuthorizationResult, 
+                accounts: accounts
+            };
+            // TODO: Evaluate whether there's any threat to not `awaiting` this expression
+            Promise.all([
+                this.#authorizationResultCache.set(authorizationResult),
+                this.#handleAuthorizationResult(authorizationResult),
+            ]);
+        } catch (e) {
+            this.#disconnect();
+            throw new WalletDisconnectedError((e instanceof Error && e?.message) || 'Unknown error', e);
+        }
+    }
+
+    #disconnect: StandardDisconnectMethod = async () => {
+        // TODO: figure out why this call throws "TypeError: _a.terminateSession is not a function"
+        //  even though the session termination is actually executed (websocket closes). 
+        try { this.#wallet?.terminateSession(); } catch (e) {}
```

Comment:
hmm good thought, let me investigate 🦺

Reply:
fixed! 

I still don't love this solution of using the wallet proxy request handler to handle what is not an MWA RPC request (`terminateSession`). But for now I think its the best way to add this functionality - I am thinking about a big refactor to the js impl to clean this up. the terminate session API is not exposed directly to consumers so it will be safe to refactor the implementation later on.